js/code.js

```diff
@@ -111,31 +183,39 @@ function doLogout()
 
 function addColor()
 {
-	let newColor = document.getElementById("colorText").value;
+	let firstName = document.getElementById("firstName").value;
+  let lastName = document.getElementById("lastName").value;
+  let email = document.getElementById("email").value;
+  let phone = document.getElementById("phone").value;
+
 	document.getElementById("colorAddResult").innerHTML = "";
 
-	let tmp = {user:1, firstName:newColor, lastName:"Parker", email:"1@yahoo.com", phone:3333333333 };
-	let jsonPayload = JSON.stringify( tmp );
+	if (firstName!="" && lastName!="" && email!="" && phone!="") {

```

Comment:
This is the line that checks that all fields are filled **before adding** a new contact @NicholasCG